src/domains/checkout-app/api/fragments/_fragment_feature_flag_filter.tpl.xml

```diff
@@ -1,10 +1,17 @@
 <fragment>
     <set-variable name="requestIpAddress" value="@(context.Request.Headers.GetValueOrDefault("X-Forwarded-For",""))" />
-    <set-variable name="allowedIpsEnabledAuthentication" value="{{enable-checkout-authentication-ip-whitelist}}" />
-    <!-- example of more feature flags -->
-    <!-- <set-variable name="allowedIpsFeature_1" value="{{Feature1IpWhitelist}}" /> -->
-    <!-- <set-variable name="allowedIpsFeature_2" value="{{Feature2IpWhitelist}}" /> -->
-    <!-- <set-variable name="allowedIpsFeature_3" value="{{Feature3IpWhitelist}}" /> -->
+    <set-variable name="featureFlagsStatusMap" value="{{checkout-feature-flag-map}}" />
+
+
+    <!-- Parse the JSON feature flag status map -->
+    <set-variable name="enableAuthIpWhiteList" value="@{
+        var featureFlagsStatusMap = JsonConvert.DeserializeObject<Dictionary<string, string>>(context.Variables.GetValueOrDefault("featureFlagsStatusMap", "{}"));
+        return featureFlagsStatusMap.GetValueOrDefault("enableAuthIpWhiteList", "-");
+    }" />
+    <set-variable name="enablePspPickerPageIpWhiteList" value="@{
+        var featureFlagsStatusMap = JsonConvert.DeserializeObject<Dictionary<string, string>>(context.Variables.GetValueOrDefault("featureFlagsStatusMap", "{}"));
+        return featureFlagsStatusMap.GetValueOrDefault("enablePspPickerPageIpWhiteList", "-");
+    }" />
```

Comment:
could we move this code in the `checkout-feature-flag` set-variable tag? 
those variables are used only here. this way we could perform Json deserialization once instead of replicating it for each variable